src/OpenFeature/ProviderRepository.cs

```diff
@@ -256,9 +261,9 @@ public FeatureProvider GetProvider()
             }
         }
 
-        public FeatureProvider GetProvider(string clientName)
+        public FeatureProvider GetProvider(string? clientName)
         {
-            if (string.IsNullOrEmpty(clientName))
+            if (clientName == null || string.IsNullOrEmpty(clientName))
```

Comment:
Looks like maybe this check got added when there was already a `IsNullOrEmpty` check. Or the intent here is unclear.

Reply:
It is a strange change, but only the newer dotnet SDK has the proper null check.
See the error here:
![Screenshot 2024-04-03 at 08 40 06](https://github.com/open-feature/dotnet-sdk/assets/2493377/684cc019-393a-4320-b70d-a6d52d3afecd)
![Screenshot 2024-04-03 at 08 38 47](https://github.com/open-feature/dotnet-sdk/assets/2493377/b77c7d83-92b9-4e3f-8bd7-f8f371f02af1)

Reply:
Ok. Apparently .netstandard and .net framework don't have nullable annotations for IsNullOrEmpty.

In cases like this I think we need a comment to indicate that. Otherwise it is something everyone will want to optimize away

Reply:
I added a target framework conditional, so it is way more explicit why that null check needs to be there. This way, the next developer will know why! :)

I hope it addresses your concerns!